core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -782,6 +784,7 @@ private void addPendingDelete(Index index, PendingDelete pendingDelete) {
                 pendingDeletes.put(index, list);
             }
             list.add(pendingDelete);
+            numUncompletedDeletes.incrementAndGet();
```

Comment:
😢  OK.. can we add comments on this and the corresponding public method to say we need it for testing and why?